Tests/MongoSwiftSyncTests/ClientSessionTests.swift

```diff
@@ -281,55 +272,37 @@ final class SyncClientSessionTests: MongoSwiftTestCase {
             try collection.insertOne(["x": BSON(x)], session: session)
         }
 
-        var id: BSON?
-        var seenFind = false
-        var seenGetMore = false
-
-        let center = NotificationCenter.default
-        let observer = center.addObserver(forName: .commandStarted, object: nil, queue: nil) { notif in
-            guard let event = notif.userInfo?["event"] as? CommandStartedEvent else {
-                return
-            }
-
-            if event.command["find"] != nil {
-                seenFind = true
-                if let id = id {
-                    expect(id).to(equal(event.command["lsid"]))
-                } else {
-                    expect(event.command["lsid"]).toNot(beNil())
-                    id = event.command["lsid"]
-                }
-            } else if event.command["getMore"] != nil {
-                seenGetMore = true
-                expect(id).toNot(beNil())
-                expect(event.command["lsid"]).toNot(beNil())
-                expect(event.command["lsid"]).to(equal(id))
-            }
-        }
-
         // explicit
-        id = .document(session.id!)
-        seenFind = false
-        seenGetMore = false
+        monitor.beginMonitoring()
         let cursor = try collection.find(options: FindOptions(batchSize: 2), session: session)
         expect(cursor.next()).toNot(beNil())
         expect(cursor.next()).toNot(beNil())
         expect(cursor.next()).toNot(beNil())
-        expect(seenFind).to(beTrue())
-        expect(seenGetMore).to(beTrue())
+        monitor.stopMonitoring()
```

Comment:
seems like we turn monitoring on/off and clear out events a lot, do you think something like a `withMonitoring` method that allows you to run a closure would be useful?